packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
+      //TODO: how to generate proper ID?
+      const id = 'AdditionalSecretRotation';
+      return this.newSecretRotation(id, options);
```

Comment:
I believe we're missing calling `options.secret.attach(this)` in this method.

Reply:
This is not the same behavior CDK has in multi-user rotation (it expects the secret to be attached already). Related to #13719. What do you think?